### PR TITLE
Fix(eos_designs): Use CP-Profile for WAN HA when DP-Profile is not configured

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Use CP-Profile for WAN HA when DP-Profile is not configured

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

When data_plane is not configured for IPsec for WAN, use control_plane profile instead

## How to test

molecuke

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_designs/structured_config/overlay/ip_security.py [hunk 1]
```diff
@@ -45,8 +44,9 @@ def ip_security(self: AvdStructuredConfigOverlay) -> dict | None:
         return strip_null_from_data(ip_security)
 
     def _append_data_plane(self: AvdStructuredConfigOverlay, ip_security: dict, data_plane_config: dict) -> None:
-        """
-        In place update of ip_security
+        """In place update of ip_security for Data Plane.
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/structured_config/overlay/ip_security.py]:
```suggestion
        """
        In place update of ip_security for Data Plane.
```

**Review by ClausHolbechArista — DISMISSED** [review]
Add not in release notes with this behavioral change.

File: python-avd/pyavd/_eos_designs/structured_config/overlay/ip_security.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def ip_security(self: AvdStructuredConfigOverlay) -> dict | None:
         return strip_null_from_data(ip_security)
 
     def _append_data_plane(self: AvdStructuredConfigOverlay, ip_security: dict, data_plane_config: dict) -> None:
-        """In place update of ip_security."""
+        """In place update of ip_security for DataPlane."""
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/structured_config/overlay/ip_security.py:50]: Line 42 and 43 in this file already implemented the behavior advertised here as "new".

File: python-avd/pyavd/_eos_designs/structured_config/overlay/router_path_selection.py [hunk 3]
```diff
@@ -45,9 +45,13 @@ def _cp_ipsec_profile_name(self: AvdStructuredConfigOverlay) -> str:
...
     @cached_property
     def _dp_ipsec_profile_name(self: AvdStructuredConfigOverlay) -> str:
-        """Returns the IPsec profile name to use for Data-Plane."""
-        # TODO: need to use CP one if 'wan_ipsec_profiles.data_plane' not present
-        return get(self._hostvars, "wan_ipsec_profiles.data_plane.profile_name", default="DP-PROFILE")
+        """Returns the IPsec profile name to use for Data-Plane.
+
+        If no data-plane config is present for IPsec, default to _cp_ipsec_profile_name
+        """
+        if (data_plane := get(self._hostvars, "wan_ipsec_profiles.data_plane")) is not None:
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_designs/structured_config/overlay/router_path_selection.py:52]: It seems like this is the only place we are actually changing any behavior. And to me this seems more like a bugfix. The name here is only used for the `ha_path_group["ipsec_profile"]` field.

Comment (gmuloc) [reply python-avd/pyavd/_eos_designs/structured_config/overlay/router_path_selection.py:52]: yeah this is the TODO on the left

File: python-avd/pyavd/_eos_designs/structured_config/overlay/ip_security.py (hunk 2 above)
Comment (gmuloc) [reply python-avd/pyavd/_eos_designs/structured_config/overlay/ip_security.py:50]: yeah I picked up the TODO in the other file and that's probably why I thought this was new - let me change the title

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg [hunk 4]
```diff
@@ -308,8 +289,16 @@ ip security
       shared-key 7 0007054B145A1F0E0928424A0C0B4812160C09551511170B121907214A333B286214687C782720215B0B67637B7B666B3873293274733B31233B6D2A332315696A
       dpd 10 60 clear
    !
+   profile ONE-PROFILE-TO-CONTROL-THEM-ALL
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg]: This test should be done on a device with HA to reflect the actual behavioral change.

Comment (gmuloc) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg]: yup this is on my todo list - will move this to draft

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md [hunk 5]
```diff
@@ -196,6 +196,11 @@ With AVD version 5.0.0 the valid values for `wan_mode` key are now `legacy-autov
 
 See the [porting guide](../porting-guides/5.x.x.md#wan-mode-autovpn-renamed-wan-mode-legacy-vpn) for details.
 
+#### Use Control Plane IPsec profile for WAN HA when Data Plane profile is not configured
+
+For `wan_ipsec_profiles`, AVD now uses the `control_plane` profile for WAN HA Data
+Plane when the `data_plane` profile is not configured.
+
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/release-notes/5.x.x.md]: No need to add it here. It was always the case, but just buggy for the HA case.

```suggestion
```

**Review by ClausHolbechArista — APPROVED** [review]

**Review by Vibhu-gslab — APPROVED** [review]

**Pull request merged**